File: src/service/segmentService.ts
import { Document } from "@langchain/core/documents";
import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";

class SegmentService {
  private textSplitter: RecursiveCharacterTextSplitter | null = null;
  constructor() {
    this.textSplitter = new RecursiveCharacterTextSplitter({
      chunkSize: 768,
      chunkOverlap: 256,
    });
  }

  async splitText(documents: Document[]) {
    return await this.textSplitter?.splitDocuments(documents);
  }
}

const segmentService = new SegmentService();
export default segmentService;

File: fastapi/service/Segmentservice.py
from typing import List, Optional
from langchain_core.documents import Document
from langchain_text_splitters import CharacterTextSplitter


class SegmentService(object):
    """docstring for SegmentService."""

    text_splitter: CharacterTextSplitter = None

    def __init__(self):
        super(SegmentService, self).__init__()
        self.text_splitter = CharacterTextSplitter(
            separator="\n\n",
            chunk_size=768,
            chunk_overlap=256,
            length_function=len,
            is_separator_regex=False,
        )

    def splitText(
        self, texts: List[str], metadatas: Optional[List[dict]]
    ) -> List[Document]:
        texts = self.text_splitter.create_documents([texts], [metadatas])
        return texts


segmentService = SegmentService()
